pygmt/base_plotting.py

```diff
@@ -994,3 +994,368 @@ def text(
 
                 arg_str = " ".join([fname, build_arg_string(kwargs)])
                 lib.call_module("text", arg_str)
+
+    @fmt_docstring
+    @use_alias(R="region", J="projection", B="frame", C="offset")
+    @kwargs_to_strings(R="sequence",)
+    def meca(
+        self,
+        spec,
+        scale,
+        lon=None,
+        lat=None,
+        depth=None,
+        convention=None,
+        component="full",
+        plot_lon=None,
+        plot_lat=None,
+        **kwargs,
+    ):
+        """
+        Plot focal mechanisms.
+
+        Full option list at :gmt-docs:`supplements/seis/meca.html`
+
+        Note
+        ----
+            Currently, labeling of beachballs with text strings is only
+            supported via providing a file to `spec` as input.
+
+        {aliases}
+
+        Parameters
+        ----------
+        spec: dict, 1D array, 2D array, pd.DataFrame, or str
+            Either a filename containing focal mechanism parameters as columns,
+            a 1- or 2-D array with the same, or a dictionary. If a filename or
+            array, `convention` is required so we know how to interpret the
+            columns/entries. If a dictionary, the following combinations of
+            keys are supported; these determine the convention. Dictionary
+            may contain values for a single focal mechanism or lists of
+            values for many focal mechanisms. A Pandas DataFrame may
+            optionally contain columns lat, lon, depth, plot_lon,
+            and/or plot_lat instead of passing them to the meca method.
+
+            - ``"aki"`` — *strike, dip, rake, magnitude*
+            - ``"gcmt"`` — *strike1, dip1, rake1, strike2, dip2, rake2,
+              mantissa, exponent*
+            - ``"mt"`` — *mrr, mtt, mff, mrt, mrf, mtf, exponent*
+            - ``"partial"`` — *strike1, dip1, strike2, fault_type, magnitude*
+            - ``"principal_axis"`` — *t_exponent, t_azimuth, t_plunge,
+              n_exponent, n_azimuth, n_plunge, p_exponent, p_azimuth, p_plunge,
+              exponent*
+
+        scale: str
```

Comment:
One thing I noticed when I was using this code is that you cannot use the GMT options directly. So you can only plot full moment tensors (force double couple) by doing something like: 

`fig.meca(
    spec="focal_mechanisms.txt", convention='mt', scale="1", component="dc", offset="0.5p"
)`

Which is great for novice users, and is obviously more pythonic, but I feel like this would be a little frustrating for GMT vets - who still have valid reasons for wanting to use this API (for context I am not a GMT vet, but know people who are). 

Is is not possible to also allow this in addition to the keywords:

`fig.meca(
    spec="focal_mechanisms.txt", Sd='1', C="0.5p"
)` 

Or did you guys decide this was not okay in the grand scheme of things?

When I tried my hand at [this](https://github.com/sgjholt/pygmt/commit/775faa7637aa8fdc13bc372d4fa51987ca26134e)  that was the first thing I did (this version is definitely far inferior overall - but I think allowing GMT options directly is a strength).

Reply:
Hi there! Thanks for reviewing. Yes, you should be able to pass in a file to `spec`. Or an array containing the same info.

Reply:
I was so excited to get away from GMT syntax that I didn't think about making that a possibility. Shouldn't be too difficult. I'll implement this over the weekend. It's neat to see your previous work on this!

Reply:
> Is is not possible to also allow this in addition to the keywords:
> 
> fig.meca( spec="focal_mechanisms.txt", Sd='1', C="0.5p" )

PyGMT is discouraging the use of short-form arguments. See https://github.com/GenericMappingTools/pygmt/issues/262 for the reasons.